crates/autopilot/src/domain/settlement/mod.rs

```diff
@@ -40,15 +37,58 @@ impl Settlement {
 
         if persistence
             .auction_has_settlement(solution.auction_id())
-            .await?
+            .await
+            .map_err(Error::from)?
         {
             // This settlement has already been processed by another environment.
             //
             // TODO: remove once https://github.com/cowprotocol/services/issues/2848 is resolved and ~270 days are passed since bumping.
             return Err(Error::WrongEnvironment);
         }
 
-        let auction = persistence.get_auction(solution.auction_id()).await?;
+        let auction = persistence
+            .get_auction(solution.auction_id())
+            .await
+            .map_err(Error::from)?;
+
+        let winning_solution = persistence
+            .get_winning_solution(solution.auction_id())
+            .await
+            .map_err(Error::from)?;
+
+        if transaction.solver != winning_solution.solver() {
+            return Err(Error::SolverMismatch {
+                expected: winning_solution.solver(),
+                got: transaction.solver,
+            });
+        }
+
+        // only debugging for now
+        match solution.score(&auction) {
+            Ok(score) => {
+                tracing::debug!(
+                    "Settlement for auction {} has score {}",
+                    solution.auction_id(),
+                    score
+                );
+
+                if score < winning_solution.score() {
+                    tracing::warn!(
+                        "Settlement for auction {} has lower score {} than the winning solution {}",
+                        solution.auction_id(),
+                        score,
+                        winning_solution.score()
+                    );
+                }
```

Comment:
nit: print only one of the logs. Also, having `auction_id` as a log separate field would be better for kibana search.
```suggestion
                if score < winning_solution.score() {
                    tracing::warn!(
                        auction_id = ?solution.auction_id(),
                        "Settlement for auction has lower score {} than the winning solution {}",
                        score,
                        winning_solution.score()
                    );
                } else {
                    tracing::debug!(
                        auction_id = ?solution.auction_id()
                        "Settlement for auction has score {}",
                        score
                    );
                }
```

Reply:
added auction_id = ?solution.auction_id()